qiskit_experiments/library/characterization/analysis/resonator_spectroscopy_analysis.py

```diff
@@ -0,0 +1,72 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Spectroscopy analysis class for resonators."""
+
+from typing import List, Tuple
+import numpy as np
+
+import qiskit_experiments.curve_analysis as curve
+from qiskit_experiments.curve_analysis import ResonanceAnalysis
+from qiskit_experiments.framework import AnalysisResultData, ExperimentData
+from qiskit_experiments.framework.matplotlib import get_non_gui_ax
+from qiskit_experiments.data_processing.nodes import ProjectorType
+
+
+class ResonatorSpectroscopyAnalysis(ResonanceAnalysis):
+    """Class to analysis resonator spectroscopy."""
+
+    @classmethod
+    def _default_options(cls):
+        options = super()._default_options()
+        options.dimensionality_reduction = ProjectorType.ABS
+        options.result_parameters = [curve.ParameterRepr("freq", "meas_freq", "Hz")]
```

Comment:
In this case of this experiment, one is often as interested in the linewidth of the resonator as its frequency. Can we add a `kappa` result as well?

Reply:
Yes, good idea. See here: 3e258cb